lib/order.rb

```diff
@@ -1,18 +1,74 @@
+require "csv"
+
 module Grocery
   class Order
-    attr_reader :id, :products
+    attr_reader :id, :products, :orders
+    @@orders = []
 
     def initialize(id, products)
       @id = id
       @products = products
-    end
+      if @products == nil
+        @products == 0
+      end
+    end #--- end initialize method
 
     def total
-      # TODO: implement total
-    end
+      total = 0
+      @products.each do |product, price|
+        total = total + price
+      end
+      total = total + (total * 0.075).round(2)
+      return total
+    end #--- end total method
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
-    end
-  end
-end
+      if @products.keys.include?(product_name)
+        return false
+      else
+        @products[product_name] = product_price
+        return true
+      end
+    end #--- end add_product method
+
+    def remove_product(product_name)
+      if @products.keys.include?(product_name)
+        @products.delete(product_name)
+        return true
+      else
+        return false
+      end
+    end #--- end remove_product method
+
+    def self.all
+      if @@orders.length == 0
+        CSV.open("support/orders.csv", 'r').each do |line|
+          id = line[0].to_i
+          products = []
+          product_list = line[1].split(";")
+          product_list.each do |item|
+            separate = item.split(":")
+            product_hash = {}
+            product_hash[separate[0]] = separate[1].to_f
+            products << product_hash
```

Comment:
I'm not sure this is quite what you want - this will create an array of hashes of form:

```ruby
[
  { 'bananas': 1.99 },
  { 'crackers': 3.50 }
]
```

whereas the project requirements (and wave 1 tests) use just a hash, without an array:

```ruby
{
  'bananas': 1.99,
  'crackers': 3.50
}
```

These will behave differently in subtle ways.